larray_editor/arraywidget.py

```diff
@@ -536,11 +648,81 @@ class ArrayEditorWidget(QWidget):
     def __init__(self, parent, data, readonly=False, bg_value=None, bg_gradient=None, minvalue=None, maxvalue=None):
         QWidget.__init__(self, parent)
         readonly = np.isscalar(data)
-        self.model = ArrayModel(data, readonly=readonly, parent=self,
-                                bg_value=bg_value, bg_gradient=bg_gradient,
-                                minvalue=minvalue, maxvalue=maxvalue)
-        self.view = ArrayView(self, self.model, data.dtype, data.shape)
-
+        self.readonly = readonly
+
+        self.model_axes = LabelsArrayModel(parent=self, readonly=readonly)
+        self.view_axes = LabelsView(parent=self, model=self.model_axes, position=(LEFT, TOP))
+
+        self.model_xlabels = LabelsArrayModel(parent=self, readonly=readonly)
+        self.view_xlabels = LabelsView(parent=self, model=self.model_xlabels, position=(RIGHT, TOP))
+
+        self.model_ylabels = LabelsArrayModel(parent=self, readonly=readonly)
+        self.view_ylabels = LabelsView(parent=self, model=self.model_ylabels, position=(LEFT, BOTTOM))
+
+        self.model_data = DataArrayModel(parent=self, readonly=readonly, bg_value=bg_value, bg_gradient=bg_gradient,
+                                         minvalue=minvalue, maxvalue=maxvalue)
+        self.view_data = DataView(parent=self, model=self.model_data, dtype=data.dtype, shape=data.shape)
+
+        self.mapping = MappingLArrayData(axes_model=self.model_axes, xlabels_model=self.model_xlabels,
```

Comment:
I think adapter would be a better term.
self.data_adapter = LArrayDataAdapter(...)?